test/OpenTelemetry.Instrumentation.AspNetCore.Tests/RouteTests/TestApplication/TestApplicationFactory.cs

```diff
@@ -167,4 +174,26 @@ private static WebApplication CreateRazorPagesApplication()
 
         return app;
     }
+
+    private static WebApplication CreateExceptionHandlerApplication()
+    {
+        var builder = WebApplication.CreateBuilder();
+        builder.Logging.ClearProviders();
+
+#if NET7_0_OR_GREATER
+        builder.Services.AddProblemDetails();
+#endif
+
+        var app = builder.Build();
+
+#if NET7_0_OR_GREATER
+        app.UseExceptionHandler();
+#endif
```

Comment:
Unit tests are failing. I believe you may need to use the overload of [UseExceptionHandler that accepts a delegate](https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.builder.exceptionhandlerextensions.useexceptionhandler#microsoft-aspnetcore-builder-exceptionhandlerextensions-useexceptionhandler(microsoft-aspnetcore-builder-iapplicationbuilder-system-action((microsoft-aspnetcore-builder-iapplicationbuilder)))).

Also, it should be available for .NET 6+

Reply:
Done. I understand now. ProblemDetails are only available from dotnet 7 onwards